fms/utils/serialization.py

```diff
@@ -276,7 +277,7 @@ def load_state_dict(
             glob_pattern_list = [glob_pattern]
         elif source == "meta":
             glob_pattern_list = ["*.pth", "*.safetensors"]
-        elif source == "hf":
+        elif source == "hf" or "gptq_hf" in source:
```

Comment:
would it make sense to have a gptq=True/False to get_model use this adapter? If you use this adapter it will use it automatically, but if you specify hf and gptq=True, it will resolve to gptq_hf

Reply:
I'd simply remove this at this point, and use the "hf_unfused" source if needed.

Reply:
Even more, I'd make "hf_unfused" kind of private, and if you pass "unfuse_weights" and quantization="gptq" to inference.py have the "hf" source become "hf_unfused"

Reply:
at this point in the code, we check `source` to determine the extension of the ckpt to be loaded. We actually don't need the additional check for `gptq_hf` or similar: as long as `hf` is in the name it will trigger the correct ckpt selection. At some point in the past I was testing a different custom source name for gptq models, hence the spurious code. Will remove this.

Reply:
@ani300 good idea of automatically adjusting source by checking if `linear_type` is `gptq`